### PR TITLE
New package: nody-greeter-1.4.1

### PR DESCRIPTION
**Pull request by JezerM**
#### Testing the changes
- I tested the changes in this PR: **YES**

#### New package
- This new package conforms to the [quality requirements](https://github.com/void-linux/void-packages/blob/master/Manual.md#quality-requirements): **YES**

#### Local build testing
- I built this PR locally for my native architecture, (x86_64)

### CODE REVIEW

**Comment by JezerM** [comment]
node-gyp fails on aarch64 and arm because of `-m64` flag not existing on their compilers, therefore these architectures are unavailable at the moment.

File: srcpkgs/nody-greeter/INSTALL [hunk 1]
```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/env bash
```
Comment (Chocimier) [inline srcpkgs/nody-greeter/INSTALL:1]: Remove INSTALL and REMOVE, things are already done in post_install.

File: srcpkgs/nody-greeter/template [hunk 2]
```diff
@@ -0,0 +1,36 @@
+# Template file for 'nody-greeter'
+pkgname=nody-greeter
+version=1.3.1
+revision=1
+archs="i686 x86_64"
+build_style=meta
```
Comment (Chocimier) [inline srcpkgs/nody-greeter/template:6]: Remove build_style= altogether, it's not needed.

File: srcpkgs/nody-greeter/template [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+bootstrap=true
```
Comment (Chocimier) [inline srcpkgs/nody-greeter/template:7]: It isn't a bootstrap package. If you see error trying to build it first time, refer to Manual.

File: srcpkgs/nody-greeter/template [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+conf_files="/etc/lightdm/web-greeter.yml /etc/lightdm/Xgreeter /etc/xdg/lightdm/lightdm.conf.d/90-greeter-wrapper.conf"
+hostmakedepends="base-devel nodejs python3"
+makedepends="base-devel libxcb-devel libX11-devel liblightdm-gobject gobject-introspection cairo"
+depends="cairo libX11 libxcb lightdm liblightdm-gobject"
```
Comment (Chocimier) [inline srcpkgs/nody-greeter/template:11]: depends= are mostly generated automatically, see `04-generate-runtime-deps` in build log.

File: srcpkgs/nody-greeter/template [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+short_desc="LightDM greeter made with node.js"
+maintainer="JezerM <amyuki4@gmail.com>"
+license="GPL-3.0-or-later"
+homepage="https://github.com/JezerM/nody-greeter"
+distfiles="https://github.com/JezerM/nody-greeter/archive/${version}.tar.gz>${pkgname}-${version}.tar.gz"
+checksum="862a43d0f6a8de5ab071b8e961b811c15d02ed6b7bc0d3adee281d6ae9bbfedf"
+mutable_files="/usr/share/web-greeter/*"
```
Comment (Chocimier) [inline srcpkgs/nody-greeter/template:18]: Does it really alter files in usr/share after they are installed?

File: srcpkgs/nody-greeter/template [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+	node make build --DEST_DIR="${DESTDIR}"
+}
+
+do_install() {
+	node make install --DEST_DIR="${DESTDIR}"
+}
+
+post_install() {
+	rm -f ${DESTDIR}/usr/bin/nody-greeter
+	ln -sf ${DESTDIR}/opt/nody-greeter/nody-greeter ${DESTDIR}/usr/bin/nody-greeter
```
Comment (Chocimier) [inline srcpkgs/nody-greeter/template:35]: `ln -s /opt/nody-greeter/nody-greeter ${DESTDIR}/usr/bin/nody-greeter`, if anything.

**Review by Chocimier — CHANGES_REQUESTED** [review]
I am not interested in merging this, but can point out some issues to fix, in case someone else is.

> node-gyp fails on aarch64 and arm because of -m64 flag not existing on their compilers

*because build scripts pass x86_64-specific flag to all targets

File: srcpkgs/nody-greeter/template (hunk 5 above)
Comment (JezerM) [reply srcpkgs/nody-greeter/template:18]: The user could modify the files inside `/usr/share/web-greeter`, either adding, modifying or deleting themes.

**Comment by JezerM** [comment]
Added "cross-compile" option to nody-greeter build process. Different architectures than `x86_64` are now able to be cross-compiled.

Tested with `i686`, `aarch64`, `armv7l`, and `armv6l`.

**Comment by JezerM** [comment]
Updated nody-greeter to 1.4.1. I don't know if this will ever get merged, but I hope so~

**Pull request closed without merging**